src/components/RepoSubmission.tsx

```diff
@@ -38,11 +40,42 @@ const RepoSubmission = ({ isFormOpen, handleFormOpen }: RepoSubmissionProps): JS
       }
 
       setSubmitted(true);
-      sendMessage(userName, sanitizedUrl);
-      return ToastTrigger({ message: "Data Submitted", type: "success" });
+
+      if (currentUser?.access_token) {
+        const resp = await submitRepo(sanitizedUrl, currentUser.access_token);
+
+        try {
+          if (resp.status === 200) {
+            ToastTrigger({ message: "Repo submitted successfully", type: "success" });
+            // issue #409 - TODO: This sendMessage is not working
+            sendMessage(sanitizedUrl, userName);
+          }
+
+          if (resp.status === 404) {
```

Comment:
Hey, @janie-lee-developer per our conversation last week. This approach is what was needed to push this over the finish line. I will now issue #404 in a separate PR. Let me know if you have questions.

Reply:
Got it! Thank you for tagging me!!